### PR TITLE
HTTPServer: Options for returning new headers X-Content-Type-Options,  X-XSS-Protection and Strict-Transport-Security

### PR DESCRIPTION
**Pull request by kylebrandt**

### CODE REVIEW

**Comment by kylebrandt** [comment]
@bergquist can you do a pre-review on this before I iterate on this?

File: pkg/services/sqlstore/user.go [hunk 1]
```diff
@@ -533,7 +533,7 @@ func deleteUserInTransaction(sess *DBSession, cmd *m.DeleteUserCommand) error {
 	}
 
 	for _, sql := range deletes {
-		_, err := sess.Exec(sql, cmd.UserId)
+		_, err := sess.Exec(sql, cmgitd.UserId)
```
Comment (bergquist) [inline pkg/services/sqlstore/user.go]:
```suggestion
		_, err := sess.Exec(sql, cmd.UserId)
```

**Review by bergquist — APPROVED** [review]
LGTM

**Comment by bergquist** [comment]
needs to be added to the sample.ini file and docs as we talked about (just as ref before anyone else comments about it)

**Comment by torkelo** [comment]
needs to be added to sample.ini (commented out),  and defaults.ini

**Comment by DanCech** [comment]
LGTM

File: conf/defaults.ini [hunk 2]
```diff
@@ -182,6 +182,30 @@ cookie_samesite = lax
 # set to true if you want to allow browsers to render Grafana in a <frame>, <iframe>, <embed> or <object>. default is false.
 allow_embedding = false
 
+# Set to false if you want to disable http strict transport security (HSTS) response header.
+# This is only sent when HTTPS is enabled in this configuration.
+# HSTS tells browsers that the site should only be accessed using HTTPS.
+strict_transport_security = true
```
Comment (bergquist) [inline conf/defaults.ini]: I'm not sure we want to enable this by default. It's quite invasive unless the operator understands it. 

If this header is set, the browser won't allow any none HTTPS request to the same URL. Which I can see happen on internal company networks.

Comment (bergquist) [reply conf/defaults.ini]: Looks good otherwise!

**Comment by kylebrandt** [comment]
Some debate around HSTS...

Be it resolved, that default HSTS behavior is on when HTTPS is enabled.

For:
 * If HTTPS is setup, it is likely the desire that only HTTPS be used for better security

Against:
 * Could be a behavior (maybe breaking?) change for users that already have HTTPS enabled for some reason. Since this will get cached by browser, could cause unexpected problems for users when they upgrade. Not being able to access monitoring tools due to someone changing the deployment strategy for Grafana is also quite severe.

Right now we are leaning towards enabling by default when HTTPS in enabled based on feedback from our infra team (and my past-SRE bias). Anyone have opinions on this they would like to share?

**Comment by xlson** [comment]
We should not change the defaults in a patch release. But in a minor release it should be fine. These changes can be introduced in a patch release (without changing the defaults).

**Comment by torkelo** [comment]
Updated the PR title, we generate the changelog from PRs with the add to changelog label so a good descriptive title is importing describing the user impact (not the actual change). 

Also remember to hit the "Delete branch" button after merging a PR.

**Pull request merged**